heat-stack/app/routes/cases+/$analysisid.tsx

```diff
@@ -1,36 +1,171 @@
-// import { json, type DataFunctionArgs } from 'r'
 import { data, Outlet, useLoaderData } from 'react-router'
 import { Fragment } from 'react'
 import { type Route } from './+types/$analysisid.ts'
+import { prisma } from '#app/utils/db.server.ts'
+import { invariantResponse } from '@epic-web/invariant'
 
 export async function loader({ params }: Route.LoaderArgs) {
-    // // From the database:
-	// const user = await prisma.user.findFirst({
-	// 	select: {
-	// 		id: true,
-	// 		name: true,
-	// 		username: true,
-	// 		createdAt: true,
-	// 		image: { select: { id: true } },
-	// 	},
-	// 	where: {
-	// 		username: params.username,
-	// 	},
-	// })
-    
-    return data({
-        id: params.analysisid
-        })
+    const analysisId = Number(params.analysisid)
+
+    // Make sure analysisId is a valid number
+    invariantResponse(!isNaN(analysisId), 'Invalid analysis ID', { status: 400 })
+
+    // Fetch the case and all related information
+    const caseData = await prisma.case.findFirst({
+        where: {
+            analysis: {
+                some: {
+                    id: analysisId
+                }
+            }
+        },
+        include: {
+            homeOwner: true,
+            location: true,
+            analysis: {
+                where: {
+                    id: analysisId
+                },
+                include: {
+                    heatingInput: true,
+                    heatingOutput: true
+                }
+            }
+        }
+    })
+
+    // If no case found, return 404
+    invariantResponse(caseData, 'Case not found', { status: 404 })
+
+    return data({ caseData })
 }
 
-export default function Analysis () {
-	const data = useLoaderData<typeof loader>()
-    const id = data?.id
+export default function Analysis() {
+    const { caseData } = useLoaderData<typeof loader>()
```

Comment:
This may be remix syntax instead of react router 7 syntax. Switch over when possible.